### PR TITLE
Add a side-by-side toggle

### PR DESCRIPTION
**Pull request by jacekkopecky**
With this PR, DiffViewer windows have an icon in the navigation area to toggle line-by-line and side-by-side view.

The mirror icon is the best I found from among the standard codicon set. 

It's the second icon in the top-right corner of these screenshots:

Screen Shot 2023-01-15 at 14 51 42

Screen Shot 2023-01-15 at 14 51 23

### CODE REVIEW

**Comment by jacekkopecky** [comment]
@caponetto the 500ms update delay forced in `provider.ts` feels too long when doing a quick refresh (e.g. when the side-by-side toggle changes). 

I see that if it's completely removed, vscode doesn't show the "loading" message when opening a long diff file, which IIRC is the reason for the delay, but maybe we could reduce this to 100ms which seems to be enough for the "loading" message to show up?

File: package.json [hunk 1]
```diff
@@ -42,6 +42,18 @@
...
+    "commands": [
+      {
+        "command": "diffviewer.showLineByLine",
+        "title": "Show diff line by line",
+        "icon": "$(mirror)"
+      },
+      {
+        "command": "diffviewer.showSideBySide",
+        "title": "Show diff side by side",
+        "icon": "$(mirror~disabled)"
```
Comment (caponetto) [inline package.json:54]: This icon gives the impression that the command is disabled and can't be clicked, which is not true.

WDYT about using `file` icon for line-by-line and `files` icon for side-by-side?

File: src/extension/provider.ts [hunk 2]
```diff
@@ -24,14 +24,18 @@ export class DiffViewerProvider implements vscode.CustomTextEditorProvider {
...
 
-  public static register(args: DiffViewerProviderArgs): vscode.Disposable {
-    return vscode.window.registerCustomEditorProvider(DiffViewerProvider.VIEW_TYPE, new DiffViewerProvider(args), {
-      webviewOptions: {
-        retainContextWhenHidden: true,
-        enableFindWidget: true,
-      },
-      supportsMultipleEditorsPerDocument: false,
-    });
+  public static register(args: DiffViewerProviderArgs): vscode.Disposable[] {
```
Comment (caponetto) [inline src/extension/provider.ts:27]:
```suggestion
  public static registerContributions(args: DiffViewerProviderArgs): vscode.Disposable[] {
```

**Comment by caponetto** [comment]
> @caponetto the 500ms update delay forced in `provider.ts` feels too long when doing a quick refresh (e.g. when the side-by-side toggle changes).
> 
> I see that if it's completely removed, vscode doesn't show the "loading" message when opening a long diff file, which IIRC is the reason for the delay, but maybe we could reduce this to 100ms which seems to be enough for the "loading" message to show up?

@jacekkopecky 100ms looks good to me!

File: package.json (hunk 1 above)
Comment (jacekkopecky) [reply package.json:54]: I understand what you mean about disabled; I took inspiration from the whitespace icon in diff editor which uses this visual pattern. 

Anyway, the mirror icon clashes with the "split editor right" icon that vscode puts everywhere so the file/files icon looks better. I'll commit momentarily.

**Comment by jacekkopecky** [comment]
Thanks for the feedback, all done.

**Review by caponetto — APPROVED** [review]

**Pull request merged**